### ISSUE TITLE
toml-rs forbids empty keys, which are allowed by the spec

### ISSUE THREAD

**Issue by joelthelion** [alexcrichton/toml-rs#425]
The TOML spec says (https://toml.io/en/v1.0.0#keys):

_A bare key must be non-empty, but an empty quoted key is allowed (though discouraged)._

So I would expect the following program to work:

```
use toml;

fn main() {
    r#""" = 5"#.parse::<toml::Value>().unwrap();
}
```

But this is what I get:
`thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: EmptyTableKey, line: Some(0), col: 0, at: Some(0), message: "", key: [] } }', src/main.rs:4:40`

By comparison, python will happily run the following program:
```
import toml

print(toml.loads(""""" = 5"""))
```

**Comment by joelthelion** [comment]
I just noticed there is a fix in the following issue, but it was never merged: https://github.com/alexcrichton/toml-rs/pull/373

**Comment by Kixunil** [comment]
This was fixed, right?

**Comment by ehuss** [comment]
Yes, this has been fixed by #373.
